File: SupplierPortal.Web/Modules/ServerTypes/ScheduledEmail/EmailColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmailRow } from "./EmailRow";

export interface EmailColumns {
    Id: Column<EmailRow>;
    OfferSupplierId: Column<EmailRow>;
    OfferId: Column<EmailRow>;
    IsSended: Column<EmailRow>;
    IsAutoSended: Column<EmailRow>;
}

export class EmailColumns extends ColumnsBase<EmailRow> {
    static readonly columnsKey = 'ScheduledEmail.Email';
    static readonly Fields = fieldsProxy<EmailColumns>();
}

File: SupplierPortal.Web/Modules/ScheduledEmail/Email/EmailColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace SupplierPortal.ScheduledEmail.Columns;

[ColumnsScript("ScheduledEmail.Email")]
[BasedOnRow(typeof(EmailRow), CheckNames = true)]
public class EmailColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [Width(120)]
    public int OfferSupplierId { get; set; }
    [Width(100)]
    public int OfferId { get; set; }
    [Width(100)]
    public bool IsSended { get; set; }
    [Width(100)]
    public bool IsAutoSended { get; set; }
}